Common/Render/Text/draw_text_sdl.cpp

```diff
@@ -0,0 +1,476 @@
+		for (int j = 0; j < ARRAY_SIZE(fallbackFonts); j++) {
+			Path fontPath = Path(fontDirs[i]) / fallbackFonts[j];
+
+			if (File::Exists(fontPath)) {
+				TTF_Font *openedFont = TTF_OpenFont(fontPath.ToString().c_str(), 24);
+				int64_t numFaces = TTF_FontFaces(openedFont);
+
+				for (int k = 0; k < numFaces; k++) {
+					TTF_Font *fontFace = TTF_OpenFontIndex(fontPath.ToString().c_str(), 24, k);
+					std::string fontFaceName(TTF_FontFaceStyleName(fontFace));
+					TTF_CloseFont(fontFace);
+
+					if (strstr(fontFaceName.c_str(), "Medium") ||
+						strstr(fontFaceName.c_str(), "Regular"))
+					{
+						fallbackFontPaths_.push_back(std::make_pair(fontPath.ToString(), k));
+						break;
+					}
+				}
+
+				TTF_CloseFont(openedFont);
+			}
+		}	
+	}
+#else
+	// We don't have a fallback font for this platform.
+	// Unsupported characters will be rendered as squares.
+#endif
+}
+
+uint32_t TextDrawerSDL::CheckMissingGlyph(std::string& text) {
+	TTF_Font *font = fontMap_.find(fontHash_)->second;
+	UTF8 utf8Decoded(text.c_str());
+
+	uint32_t missingGlyph = 0;
+	for (int i = 0; i < text.length(); ) {
+		uint32_t glyph = utf8Decoded.next();
+		i = utf8Decoded.byteIndex();
+
+		if (!TTF_GlyphIsProvided32(font, glyph)) {
+			missingGlyph = glyph;
+			break;
+		}
+	}
+
+	return missingGlyph;
+}
+
+// If this returns true, the first font in fallbackFonts_ can be used as a fallback.
+bool TextDrawerSDL::FindFallbackFonts(uint32_t missingGlyph, int ptSize) {
+	// If we encounter a missing glyph, try to use one of the fallback fonts.
+	for (int i = 0; i < fallbackFonts_.size(); i++) {
+		TTF_Font *fallbackFont = fallbackFonts_[i];
+		if (TTF_GlyphIsProvided32(fallbackFont, missingGlyph)) {
+			fallbackFonts_.erase(fallbackFonts_.begin() + i);
+			fallbackFonts_.insert(fallbackFonts_.begin(), fallbackFont);
+			return true;
+		}
+	}
+
+	// If none of the loaded fonts can handle it, load more fonts.
+	for (int i = 0; i < fallbackFontPaths_.size(); i++) {
+		std::string& fontPath = fallbackFontPaths_[i].first;
+		int faceIndex = fallbackFontPaths_[i].second;
+
+		TTF_Font *font = TTF_OpenFontIndex(fontPath.c_str(), ptSize, faceIndex);
+
+		if (TTF_GlyphIsProvided32(font, missingGlyph)) {
+			fallbackFonts_.insert(fallbackFonts_.begin(), font);
+			fallbackFontPaths_.erase(fallbackFontPaths_.begin() + i);
+			return true;
+		} else {
+			TTF_CloseFont(font);
+		}
+	}
+
+	return false;
+}
+
+uint32_t TextDrawerSDL::SetFont(const char *fontName, int size, int flags) {
+	uint32_t fontHash = fontName && strlen(fontName) ? hash::Adler32((const uint8_t *)fontName, strlen(fontName)) : 0;
+	fontHash ^= size;
+	fontHash ^= flags << 10;
+
+	auto iter = fontMap_.find(fontHash);
+	if (iter != fontMap_.end()) {
+		fontHash_ = fontHash;
+		return fontHash;
+	}
+
+	const char *useFont = fontName ? fontName : "Roboto-Condensed.ttf";
+	const int ptSize = (int)((size + 6) / dpiScale_);
+
+	TTF_Font *font = TTF_OpenFont(useFont, ptSize);
```

Comment:
We include a version of Roboto-Condensed in the repo.  We actually intentionally use an "older" version of the font, but using a consistent version across platforms improves the chances buttons won't wrap weird for some people but not others.

Would you be against this preferring the one from g_VFS?

-[Unknown]

Reply:
It might sound silly, but we actually hit the `if (!font)` case all the time. Unless the current working directory provides a `Roboto-Consensed.ttf`, the one from g_VFS is used.

Reply:
Oh, I assumed it used a path.  Got it, that sounds good.

-[Unknown]